### ISSUE TITLE
LocalCache Implementation

### ISSUE THREAD

**Issue by derekperkins** [google/weasel#7]
I've integrated github.com/hashicorp/golang-lru as an in-memory local cache. Would you accept that as a PR?

**Comment by crhym3** [comment]
Of course. I've been looking at that lru quite some time ago, too, but never had a change to integrate it. Please, do.

**Comment by x1ddos** [comment]
So, this will require coordination between running instances. I've been thinking about it and it feels like it should be possible. After that, in-memory cache should be ok to use.

Will try to put something together.

**Comment by pehowell** [comment]
Since the GCS webhook notification is now deprecated, switching to Cloud PubSub for object change notifications would also help with coordinating the flushing of objects from the local cache. Thoughts?
